### PR TITLE
Catch KeyError when SHELL is not set

### LINKED ISSUE #203: catkin build fails if SHELL is not set
I was building some catkin packages using docker, which uses root by default. For some reason, `SHELL` was not set in the environment, which was causing `catkin build` to fail:
~~~
Traceback (most recent call last):
  File "/usr/bin/catkin", line 9, in <module>
    load_entry_point('catkin-tools==0.3.0', 'console_scripts', 'catkin')()
  File "/usr/lib/python2.7/dist-packages/catkin_tools/commands/catkin.py", line 219, in main
    sys.exit(args.main(args) or 0)
  File "/usr/lib/python2.7/dist-packages/catkin_tools/verbs/catkin_build/cli.py", line 229, in main
    ctx = Context.load(opts.workspace, opts.profile, opts, append=True)
  File "/usr/lib/python2.7/dist-packages/catkin_tools/context.py", line 168, in load
    ctx.load_env()
  File "/usr/lib/python2.7/dist-packages/catkin_tools/context.py", line 300, in load_env
    sticky_env = get_resultspace_environment(self.devel_space_abs, quiet=True)
  File "/usr/lib/python2.7/dist-packages/catkin_tools/resultspace.py", line 56, in get_resultspace_environment
    shell_path = os.environ['SHELL']
  File "/usr/lib/python2.7/UserDict.py", line 23, in __getitem__
    raise KeyError(key)
KeyError: 'SHELL'
~~~
The error can easily be reproduced with bash by calling `unset SHELL` before running `catkin build`.

There should be some error checking to see if the `SHELL` variable exists before trying to use it.

### PR DESCRIPTION
**Pull request by scpeters**
Fixes #203.

### CODE REVIEW

File: catkin_tools/resultspace.py [hunk 1]
```diff
@@ -53,7 +53,10 @@ def get_resultspace_environment(result_space_path, quiet=False, cached=True):
         )
 
     # Determine the shell to use to source the setup file
-    shell_path = os.environ['SHELL']
+    try:
+        shell_path = os.environ['SHELL']
+    except KeyError:
+        shell_path = ''
```
Comment (wjwwood) [inline catkin_tools/resultspace.py]: I think it would be better to use `get`:

```python
shell_path = os.environ.get('SHELL', '')
```

**Comment by wjwwood** [comment]
Fix looks good, minor implementation comment, and then +1

File: catkin_tools/resultspace.py (hunk 1 above)
Comment (scpeters) [inline catkin_tools/resultspace.py]: fixed and squashed

**Comment by wjwwood** [comment]
Thanks.

**Pull request merged**